common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java

```diff
@@ -139,32 +140,43 @@ public String getPrompt() {
     }
 
     @Override
-    public Uri getAuthorizationRequestAsHttpRequest() throws UnsupportedEncodingException {
-        Uri.Builder uriBuilder = Uri.parse(getAuthorizationEndpoint()).buildUpon();
-        for (Map.Entry<String, Object> entry : ObjectMapper.serializeObjectHashMap(this).entrySet()) {
-            uriBuilder.appendQueryParameter(entry.getKey(), entry.getValue().toString());
-        }
+    public Uri getAuthorizationRequestAsHttpRequest() {
+        Map<String, Object> qpMap = new HashMap<>();
+        qpMap.putAll(ObjectMapper.serializeObjectHashMap(this));
```

Comment:
I am assuming that this is the behavior for iOS, where if the qmap already has the item it is prioritized over the similar  query parameter item.

Reply:
No, actually, iOS does not do any thing on the MSAL calls. iOS only whitelisted all the extra query parameters when doing the broker calls, including authorization request and token request.